addons/GearSwap/libs/closetCleaner.lua

```diff
@@ -0,0 +1,367 @@
+
+register_unhandled_command(function(...)
+    local cmds = {...}
+    for _,v in ipairs(cmds) do
+        if S{'closetcleaner','cc'}:contains(v:lower()) then
+            setmetatable(cc.sandbox, {__index = gearswap.user_env})
+            cc.sandbox.itemsBylongName = T{}
+            cc.sandbox.itemsByName = T{}
+            cc.sandbox.inventoryGear = T{}
+            cc.sandbox.gsGear = T{}
+            for k,v in pairs(gearswap.res.items) do
+                cc.sandbox.itemsBylongName[gearswap.res.items[k].name_log:lower()] = k
+                cc.sandbox.itemsByName[gearswap.res.items[k].name:lower()] = k
+            end
+            cc.sandbox.jobs = {}
+            for k,v in pairs(gearswap.res.jobs) do
+                cc.sandbox.jobs[gearswap.res.jobs[k].ens] = k
+            end
+            if not windower.dir_exists(windower.addon_path..'report') then
+                windower.create_dir(windower.addon_path..'report')
+            end
+            local path = windower.addon_path:gsub('\\','/')
+            path = path..'report/'..player.name
+            require 'ccConfig'
+            cc.run_report(path)
+            cc.sandbox = nil
+        end
+    end
+    cc.sandbox = {}
+    cc.sandbox.windower = setmetatable({}, {__index = windower})
+    cc.sandbox.windower.register_event = functions.empty()
+    cc.sandbox.windower.raw_register_event = functions.empty()
+    cc.sandbox.windower.register_unhandled_command = functions.empty()
+    return true
+end)
+
+-- This function creates the report and generates the calls to the other functions
+function cc.run_report(path)
+    mainReportName = path..'_report.txt'
+    local f = io.open(mainReportName,'w+')
+    f:write('closetCleaner Report:\n')
+    f:write('=====================\n\n')
+    cc.export_inv(path)
+    cc.export_sets(path)
+    for k,v in pairs(cc.sandbox.inventoryGear) do
+        if cc.sandbox.gsGear[k] == nil then
+            cc.sandbox.gsGear[k] = 0
+        end
+    end
+    data = T{"Name", " | ", "Count", " | ", "Location", " | ", "Jobs Used", " | ", "Long Name"}
+    form = T{"%25s", "%3s", "%10s", "%3s", "%20s", "%3s", "%-88s", "%3s", "%60s"}
+    cc.print_row(f, data, form)
+    cc.print_break(f, form)
+    if ccDebug then
+        ignoredReportName = path..'_ignored.txt'
+        f2 = io.open(ignoredReportName,'w+')
+        f2:write('closetCleaner ignored Report:\n')
+        f2:write('=====================\n\n')
+        cc.print_row(f2, data, form)
+        cc.print_break(f2, form)
+    end
+    for k,v in cc.spairs(cc.sandbox.gsGear, function(t,a,b) return t[b] > t[a] end) do
+        if ccmaxuse == nil or v <= ccmaxuse then
+            printthis = 1
+            if not cc.job_used[k] then
+                cc.job_used[k] = " "
+            end
+            for i,s in ipairs(ccignore) do
+                if windower.wc_match(gearswap.res.items[k].en, s) or string.match(gearswap.res.items[k].en, s) then
```

Comment:
At this and other points you use `en` instead of `english` or `name` (same with `english_log` and `name_log` instead of `enl`). And see the comment above about recommending only one matching algorithm. If you go for `name` over `english`, try to do so consistently.